tests/infrastructure/downward-metrics.go

```diff
@@ -93,4 +99,95 @@ var _ = DescribeInfra("downwardMetrics", func() {
+		tests.DisableDownwardMetrics(virtClient)
+
+		DeferCleanup(func() {
+			By("re-enabling the downwardMetrics feature")
+			tests.EnableDownwardMetrics(virtClient)
+		})
+
+		vmi := libvmifact.NewFedora(via)
+		vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("checking the phase of the VMI")
+		Eventually(matcher.ThisVMI(vmi), 60*time.Second, 5*time.Second).Should(matcher.BeInPhase(v1.Pending))
+
+		By("re-enabling the downwardMetrics feature")
+		tests.EnableDownwardMetrics(virtClient)
+
+		Eventually(matcher.ThisVMI(vmi), 60*time.Second, 5*time.Second).ShouldNot(matcher.BeInPhase(v1.Pending))
+	},
+		Entry("using a disk", libvmi.WithDownwardMetricsVolume("vhostmd")),
+		Entry("using a virtio serial device", libvmi.WithDownwardMetricsChannel()),
+	)
+
+	DescribeTable("should update the VMI conditions if the feature is disabled", func(via libvmi.Option) {
+		vmi := libvmifact.NewFedora(via)
+		vmi = libvmops.RunVMIAndExpectLaunch(vmi, 180)
+		Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+		By("disabling downwardMetrics feature")
+		tests.DisableDownwardMetrics(virtClient)
+
+		DeferCleanup(func() {
+			By("re-enabling the downwardMetrics feature")
+			tests.EnableDownwardMetrics(virtClient)
+		})
+
+		// Triggering an update to force the reconciliation loop. This will avoid to wait much time.
+		Expect(console.LoginToFedora(vmi)).To(Succeed())
+		By("checking the VMI conditions")
+		Eventually(matcher.ThisVMI(vmi), 60*time.Second, 5*time.Second).Should(matcher.HaveConditionFalse(v1.VirtualMachineInstanceOutOfSyncConfiguration))
+	},
+		Entry("using a disk", libvmi.WithDownwardMetricsVolume("vhostmd")),
+		Entry("using a virtio serial device", libvmi.WithDownwardMetricsChannel()),
+	)
+
+	DescribeTable("should update VM conditions if the feature is disabled", func(via libvmi.Option) {
+		vm := libvmi.NewVirtualMachine(libvmifact.NewFedora(via))
+
+		By("disabling downwardMetrics feature")
+		tests.DisableDownwardMetrics(virtClient)
+
+		DeferCleanup(func() {
+			By("re-enabling the downwardMetrics feature")
+			tests.EnableDownwardMetrics(virtClient)
+		})
+
+		vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("checking the VM conditions when the VM is stopped")
+		Eventually(func(g Gomega) {
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			cond := controller.NewVirtualMachineConditionManager().GetCondition(vm, v1.VirtualMachineFailure)
+			g.Expect(cond).ToNot(BeNil())
+			g.Expect(cond.Status).To(Equal(k8sv1.ConditionTrue))
+			g.Expect(cond.Reason).To(Equal(controller.FeatureNotEnabled))
+			g.Expect(cond.Message).To(ContainSubstring("DownwardMetrics feature is not enabled"))
```

Comment:
Use gstruct matcher as you did in other tests?

Reply:
Done.